pcweb/flexdown.py

```diff
@@ -0,0 +1,147 @@
+    },
+    "code": {
+        "color": "#1F1944",
+        "bg": "#EAE4FD",
+    },
+    "pre": {
+        "border_radius": styles.DOC_BORDER_RADIUS,
+        "background": "transparent",
+        "code_tag_props": {
+            "style": {
+                "fontFamily": "inherit"
+            }
+        }
+    }
+}
+
+def md(source: str):
+    return rx.markdown(
+        source,
+        custom_styles=custom_styles,
+    )
+
+def parse_markdown_front_matter(markdown_content):
+    # Define the regular expression pattern to match front matter
+    pattern = r"^---\s*\n(.+?)\n---\s*\n(.*)$"
+    # Extract the front matter and content using the pattern
+    match = re.match(pattern, markdown_content, re.DOTALL)
+    if match:
+        front_matter = match.group(1)
+        content = match.group(2)
+        # Parse the front matter as YAML
+        try:
+            front_matter_data = yaml.safe_load(str(front_matter))
+        except yaml.scanner.ScannerError:
+            front_matter_data = front_matter
+
+        if isinstance(front_matter_data, str):
+            front_matter_data = front_matter
+
+        return front_matter_data, content
+    else:
+        return None, markdown_content
+
+def evaluate_template_string(input_string, scope):
+    # Regular expression to match the template placeholders
+    template_regex = r"{(.*?)}"
+    matches = re.findall(template_regex, input_string)
+
+    for match in matches:
+        try:
+            # Evaluate the Python expression and replace the template placeholder
+            eval_result = str(eval(match, scope))
+            input_string = input_string.replace("{" + match + "}", eval_result)
+        except Exception as e:
+            # If the evaluation fails, leave the template placeholder unchanged
+            print(f"Failed to evaluate expression '{match}': {e}")
+
+    return input_string
+
+def parse(source: str, md=md):
+    """Parse out code blocks annotated with ```reflex
+    and replace them with the output of the code block.
+    The surrounding Markdown should be left untouched.
+    """
+    front_matter, source = parse_markdown_front_matter(source)
+    if isinstance(front_matter, str):
+        exec(front_matter)
+
+    lines = source.split("\n")
+    output = []
+    in_reflex_block = False
+    current_block = []
+
+    for line in lines:
+        if not in_reflex_block:
+            if line == "" and not in_reflex_block:
+                # End normal block.
+                output.append(md("\n".join(current_block)))
+                current_block = []
+
+        if line.startswith("```reflex"):
+            line = line[len("```reflex") :]
+            in_reflex_block = True
+        elif in_reflex_block and line.startswith("```"):
+            # End reflex block.
+            in_reflex_block = False
+            try:
+                result = eval("\n".join(current_block))
+            except Exception as e:
+                print(f"Error in reflex block: {str(e)}")
```

Comment:
File and line number would be extremely helpful here

```python
raise FlexdownParseError(lineno=lineno) from e
```

Then down below in `read`, you can catch the exception and print a helpful error message with the filename, line, and original exception.

Reply:
Will add this in a follow-up, created this ticket: https://github.com/reflex-dev/reflex-web/issues/208